### PR TITLE
[Main Course] - ResNet demo

### PR DESCRIPTION
**Pull request by jz263**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    valid_data = train_datagen.flow_from_directory(valid_path, target_size=img_size, classes=None)
+    return train_data, valid_data
+
+def call_back():
+    early_stopping = EarlyStopping(verbose=1, patience=10, monitor='val_loss')
+    model_checkpoint = ModelCheckpoint(filepath='102flowersmodel.h5', verbose=1, save_best_only=True, monitor='val_loss')
+    callbacks = [early_stopping, model_checkpoint]
+    return callbacks
+
+train_path = 'dataset/flower_data_10/train'
```
Comment (Tina0330) [inline docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:120]: Jiajie, ResNet input data 的数据类型是png类型吗？ 感觉如果可以提供input & output 具体的例子，是不是会更直观一点？

**Review by Tina0330 — APPROVED** [review]
LGTM, thanks!

File: docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+$$
+\begin{gathered}
+z^{[l+1]}=W^{[l+1]} a^{[l]}+b^{[l+1]} \\
+a^{[l+1]}=g\left(z^{[l+1]}\right) \\
+z^{[l+2]}=W^{[l+2]} a^{[l+1]}+b^{[l+2]} \\
+a^{[l+2]}=g\left(z^{[l+2]}+a^{[l]}\right)
+\end{gathered}
+$$
+
+假设残差块内部选用 ReLU 作为激活函数，当发生梯度消失时，W≈0，b≈0，则有：
```
Comment (Realvincentyuan) [inline docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:29]: 梯度消失时权重会停止更新，为什么这里写`W≈0，b≈0`，是准确的吗？

Comment (Realvincentyuan) [reply docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:29]: 是想表达神经元挂彩的情况吗？即无法激活的情况

**Review by Realvincentyuan — COMMENTED** [review]
有些问题

File: docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+所以这种构造方式保证了深层的网络比浅层包含了更多（至少恒等）的图像信息。多个残差块推挤在一起，便形成了一个残差网络。
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/ResNet-Paper.png"/>
+  <figcaption>残差网络和普通深度神经网络对比</figcaption>
+</figure>
+
+## 3 用 ResNet 构造分类模型
+
+在下列 demo 中，我们使用 keras 已有的 ResNet50 与训练模型，对 Oxford 102 Flowers 数据集中的 10 种花卉图片进行多分类任务模型的构造。在工程上我们只需要修改 ResNet50 顶部的全连接层，对输入的图片数据进行裁剪，旋转，放大等数据增强，训练顶层参数即可。代码如下：
```
Comment (jhqian0909) [inline docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:55]: ResNet50 与训练模型

Comment (jhqian0909) [reply docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:55]: ‘训练顶层参数’ ResNet50的参数应该也要参与训练的吧

File: docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md (hunk 2 above)
Comment (jz263) [reply docs/Main_Course/Deep_Learning/2021-11-04-Deep-Learning-101-ResNet.md:29]: 这里指左边两项的梯度消失，但因为有a[l]的存在，这一层的梯度没有消失

**Pull request merged**